### PR TITLE
John conroy/transformation errors

### PR DESCRIPTION
**Pull request by john-conroy**
Adds `transformation_errors` so we can easily determine for which datasets empty json was returned by the soft assay endpoint.

### CODE REVIEW

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py [hunk 1]
```diff
@@ -22,8 +22,9 @@ def _get_assay_details(doc, transformation_resources):
         raise
     json = response.json()
     if not json:
-        logger.info(f"No soft assay information returned for dataset ${uuid}.")
-        return {'assaytype': 'unknown', 'description': dataset_type, 'vitessce-hints': ['unknown-assay']}
+        empty_error_msg = 'No soft assay information returned.'
+        logger.info(f"${empty_error_msg} For dataset ${uuid}.")
+        return {'description': dataset_type, 'vitessce-hints': ['unknown-assay'], 'error': empty_error_msg}
     return json
```
Comment (lchoy) [inline src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:28]: Lines 23-28 should be moved inside the try block as response.json() will raise a [JSONDecodeError](https://requests.readthedocs.io/en/latest/api/#requests.JSONDecodeError) if the response format is not valid json.

File: src/hubmap_translation/addl_index_transformations/portal/tests/test_transform.py [hunk 2]
```diff
@@ -148,7 +148,8 @@ def raise_for_status(self):
 
 expected_output_doc_unknown_assay = expected_output_doc | {'mapped_data_types': [
     'RNAseq [Salmon]'], 'assay_display_name': [
-    'RNAseq [Salmon]'], 'visualization': False, 'vitessce-hints': ['unknown-assay']}
+    'RNAseq [Salmon]'], 'visualization': False, 'vitessce-hints': ['unknown-assay'],
+    'transformation_errors': ['No soft assay information returned.']}
```
Comment (lchoy) [inline src/hubmap_translation/addl_index_transformations/portal/tests/test_transform.py:152]: The soft-assay endpoint returns an empty json for this [example (58ebb89caf1512e9452d1f9e0e1efa8e)](https://ingest-api.dev.hubmapconsortium.org/assaytype/58ebb89caf1512e9452d1f9e0e1efa8e). Maybe fall back to dataset_type from entity? In this case, we'd end up with 'ATACseq'.

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py (hunk 1 above)
Comment (john-conroy) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:28]: `{}` shouldn't raise an error.

File: src/hubmap_translation/addl_index_transformations/portal/tests/test_transform.py (hunk 2 above)
Comment (john-conroy) [reply src/hubmap_translation/addl_index_transformations/portal/tests/test_transform.py:152]: I'm not sure I understand the question. This PR already defaults to set the `assay_display_ name` to whatever the `dataset_type` is.

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py [hunk 3]
```diff
@@ -37,6 +38,10 @@ def add_assay_details(doc, transformation_resources):
         doc['mapped_data_types'] = [assay_details.get('description')]
```
Comment (lchoy) [inline src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:38]: If the soft-assay endpoint returns an empty json for this [example (58ebb89caf1512e9452d1f9e0e1efa8e)](https://ingest-api.dev.hubmapconsortium.org/assaytype/58ebb89caf1512e9452d1f9e0e1efa8e), do we end up with `[None]`?

```suggestion
        doc['mapped_data_types'] = [assay_details.get('description', doc.get('dataset_type'))]
```

**Review by lchoy — CHANGES_REQUESTED** [review]

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py (hunk 3 above)
Comment (john-conroy) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:38]: Empty json triggers line 24 which ultimately returns `{'description': dataset_type, 'vitessce-hints': ['unknown-assay'], 'error': empty_error_msg}`.

**Review by lchoy — APPROVED** [review]

**Review by yuanzhou — APPROVED** [review]

**Comment by yuanzhou** [comment]
@john-conroy @lchoy do we need a full reindex on DEV for this PR?

**Comment by lchoy** [comment]
> @john-conroy @lchoy do we need a full reindex on DEV for this PR?

@yuanzhou No, not for this one. We have another PR coming soon to add new fields. A full reindex would be needed then.

**Pull request merged**